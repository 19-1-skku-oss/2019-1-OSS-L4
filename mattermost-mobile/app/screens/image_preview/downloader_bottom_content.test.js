// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import DownloaderBottomContent from './downloader_bottom_content';

describe('DownloaderBottomContent', () => {
    const baseProps = {
        progressPercent: 10,
        isVideo: false,
        saveToCameraRoll: true,
    };

    test('should match snapshot for downloading', () => {
        const wrapper = shallow(
            <DownloaderBottomContent {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for downloaded file and isVideo', () => {
        const wrapper = shallow(
            <DownloaderBottomContent
                {...baseProps}
                isVideo={true}
                progressPercent={100}
            />,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for downloaded file and is not video', () => {
        const wrapper = shallow(
            <DownloaderBottomContent
                {...baseProps}
                progressPercent={100}
            />,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for downloaded file', () => {
        const wrapper = shallow(
            <DownloaderBottomContent
                {...baseProps}
                saveToCameraRoll={false}
                progressPercent={100}
            />,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
