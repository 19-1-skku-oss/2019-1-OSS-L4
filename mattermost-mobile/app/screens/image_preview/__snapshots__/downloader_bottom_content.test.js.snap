// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloaderBottomContent should match snapshot for downloaded file 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 10,
    }
  }
>
  <FormattedText
    defaultMessage="Download complete"
    id="mobile.downloader.complete"
    style={
      Object {
        "color": "white",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  />
</View>
`;

exports[`DownloaderBottomContent should match snapshot for downloaded file and is not video 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 10,
    }
  }
>
  <FormattedText
    defaultMessage="Image Saved"
    id="mobile.downloader.image_saved"
    style={
      Object {
        "color": "white",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  />
</View>
`;

exports[`DownloaderBottomContent should match snapshot for downloaded file and isVideo 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 10,
    }
  }
>
  <FormattedText
    defaultMessage="Video Saved"
    id="mobile.downloader.video_saved"
    style={
      Object {
        "color": "white",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  />
</View>
`;

exports[`DownloaderBottomContent should match snapshot for downloading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 10,
    }
  }
>
  <FormattedText
    defaultMessage="Downloading..."
    id="mobile.downloader.downloading"
    style={
      Object {
        "color": "white",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  />
</View>
`;
